### ISSUE TITLE
The "data truncated" warning message is not properly formatted

### ISSUE THREAD

**Issue by tangenta** [pingcap/tidb#25076]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
drop table if exists tbl_15;
create table tbl_15 (col_90 enum('Alice','Bob','Charlie','David'));
insert ignore into tbl_15 values ('oAtXDXThAboyRjgW');
show warnings;
```

### 2. What did you expect to see? (Required)

MySQL:
```
+---------+------+---------------------------------------------+
| Level   | Code | Message                                     |
+---------+------+---------------------------------------------+
| Warning | 1265 | Data truncated for column 'col_90' at row 1 |
+---------+------+---------------------------------------------+
```

### 3. What did you see instead (Required)

```
+---------+------+------------------------------------------+
| Level   | Code | Message                                  |
+---------+------+------------------------------------------+
| Warning | 1265 | Data truncated for column '%s' at row %d |
+---------+------+------------------------------------------+
```

### 4. What is your TiDB version? (Required)

master
```
commit b7e448aa109c4ebf819ad0ca9fd6d3864d401ad5
Author: Howie <haoweili35@gmail.com>
Date:   Wed Jun 2 15:56:26 2021 +0800

    ddl: add auto random && shard_row_id_bits compatibility for temporary table (#24940)
```

**Comment by Defined2014** [comment]
I think the root cause is same as https://github.com/pingcap/tidb/pull/27959#issuecomment-931879368

**Comment by dveeden** [comment]
Looks like this was fixed by #47267

```
sql> drop table if exists tbl_15;
Query OK, 0 rows affected, 1 warning (0.0015 sec)
Note (code 1051): Unknown table 'test.tbl_15'

sql> create table tbl_15 (col_90 enum('Alice','Bob','Charlie','David'));
Query OK, 0 rows affected (0.0162 sec)

sql> insert ignore into tbl_15 values ('oAtXDXThAboyRjgW');
Query OK, 1 row affected, 1 warning (0.0007 sec)
Warning (code 1265): Data truncated for column 'col_90' at row 1

sql> show warnings;
+---------+------+---------------------------------------------+
| Level   | Code | Message                                     |
+---------+------+---------------------------------------------+
| Warning | 1265 | Data truncated for column 'col_90' at row 1 |
+---------+------+---------------------------------------------+
1 row in set (0.0008 sec)

sql> SELECT TIDB_VERSION()\G
*************************** 1. row ***************************
TIDB_VERSION(): Release Version: v7.5.0-alpha-6-g2992c1cc98
Edition: Community
Git Commit Hash: 2992c1cc98bfee3b62a27b9913a308e1cfa96667
Git Branch: master
UTC Build Time: 2023-09-28 09:13:58
GoVersion: go1.21.0
Race Enabled: false
Check Table Before Drop: false
Store: unistore
1 row in set (0.0010 sec)
```
